scalac-scoverage-plugin/src/main/scala/scoverage/report/ScoverageXmlWriter.scala

```diff
@@ -7,8 +7,15 @@ import _root_.scoverage._
 import scala.xml.{Node, PrettyPrinter}
 
 /** @author Stephen Samuel */
-class ScoverageXmlWriter(sourceDir: File, outputDir: File, debug: Boolean) {
+class ScoverageXmlWriter(sourceDirectories: Seq[File], outputDir: File, debug: Boolean) {
 
+  def this (sourceDir: File, outputDir: File, debug: Boolean) {
+    this(Seq(sourceDir), outputDir, debug);
+  }
+
+  // Source paths in canonical form WITH trailing file separator
+  val formattedSourcePaths: Seq[String] = sourceDirectories filter ( _.isDirectory ) map ( _.getCanonicalPath + File.separator )
```

Comment:
I wanted to make as small changes as possible required. This can be refactored of course.